api-service/webapp-demo/src/app/app.module.ts

```diff
@@ -39,6 +40,28 @@ import {CdkTableModule} from '@angular/cdk/table';
 import {HttpClientModule} from '@angular/common/http';
 
 import { AppComponent } from './app.component';
+import { TestJobComponent } from './test_job.component';
+import { CrowdSourceApiService} from './crowd_source_api.service';
+
+const appRoutes: Routes = [
+  {
+    path: 'test_job',
+    component: TestJobComponent
+  },
+  {
+    path: 'test_job/:jobId',
+    component: TestJobComponent
+  },
+  {
+    path: 'test_job/:jobId/:questionId',
+    component: TestJobComponent
+  },
+  {
+    path: '',
+    redirectTo: '/test_job',
+    pathMatch: 'full'
```

Comment:
Just to double check that I understand how this will work, in the future, we'll replace `test_job` with `passive_toxicity` or `identity_hate` or whatever labeling job we have, right? It seems a little redundant that for each labeling job we do, there are 3 routes that map to the same component. But no need to fix that now, just curious if you think that's ok or if we can make one route per job with optional parameters?

Reply:
Yeah, that's the idea, making one route per component. And you have a point re: the optional params; making them route params vs optional params is a design decision, angular guide recommends route params for things that we need to render the component and optional params for things we can do without. I think we should consider jobId to be required (since we technically need that to know where we're getting the question from, if I understand the code correctly and then drop the plain test_job route without jobId) and maybe allow questionId can be optional. What do you think? (Referring to the guide here: https://angular.io/guide/router#route-parameters-required-or-optional). Or they could both be optional if that's your preference! I'll add a TODO, but I'd like to hear your thoughts.

I've been thinking about this a bit more, and I think we'll want to give a bit more thought in designing this such that there's minimal code copied between components. I think there should be more refactoring of TestJobComponent, possibly to create a root component that the other components can inherit from, but I want to think it over a bit more.